### DISCUSSION TITLE
EmbeddedChannel - write - duplicates the buffer?

### DISCUSSION THREAD

**Discussion by ShlomiRex** [netty/netty#12793] [General]
I have handler that uses `EmbeddedChannel` but NOT in a test. It uses embedded channel to call decoder right away. So, the pipeline is not modified.

Here is example code:
https://github.com/ShlomiRex/netty-multiplex-two-protocols/blob/e83f83c3bf4643c139468406a016ecb5244fabae/src/main/java/handlers/ProtocolHandler.java#L36

My question is, does writing to embedded channel duplicates the `msg` buffer? What happens if the `msg` is a large buffer? I don't want to duplicate the bytes.

**Comment by chrisvest** [comment]
`EmbeddedChannel` itself does not duplicate or copy the buffer. It uses retain/release.
